File: src/extras-kunitconversion/libvalue.cpp
#define WORKAROUND_INNER_CLASS_DEFINITION_KUnitConversion__Unit
#define WORKAROUND_INNER_CLASS_DEFINITION_KUnitConversion__Value
#include <QChar>
#include <QString>
#include <QByteArray>
#include <cstring>
#include <QVariant>
#include <value.h>
#include "libvalue.h"
#include "libvalue.hxx"

KUnitConversion__Value* KUnitConversion__Value_new() {
    return new KUnitConversion::Value();
}

KUnitConversion__Value* KUnitConversion__Value_new2(double number, const KUnitConversion__Unit* unit) {
    return new KUnitConversion::Value(static_cast<qreal>(number), *unit);
}

KUnitConversion__Value* KUnitConversion__Value_new3(double number, const libqt_string unitString) {
    QString unitString_QString = QString::fromUtf8(unitString.data, unitString.len);
    return new KUnitConversion::Value(static_cast<qreal>(number), unitString_QString);
}

KUnitConversion__Value* KUnitConversion__Value_new4(double number, int unitId) {
    return new KUnitConversion::Value(static_cast<qreal>(number), static_cast<KUnitConversion::UnitId>(unitId));
}

KUnitConversion__Value* KUnitConversion__Value_new5(const QVariant* number, const libqt_string unitString) {
    QString unitString_QString = QString::fromUtf8(unitString.data, unitString.len);
    return new KUnitConversion::Value(*number, unitString_QString);
}

KUnitConversion__Value* KUnitConversion__Value_new6(const KUnitConversion__Value* other) {
    return new KUnitConversion::Value(*other);
}

void KUnitConversion__Value_OperatorAssign(KUnitConversion__Value* self, const KUnitConversion__Value* other) {
    self->operator=(*other);
}

void KUnitConversion__Value_Swap(KUnitConversion__Value* self, KUnitConversion__Value* other) {
    self->swap(*other);
}

bool KUnitConversion__Value_OperatorEqual(const KUnitConversion__Value* self, const KUnitConversion__Value* other) {
    return (*self == *other);
}

bool KUnitConversion__Value_OperatorNotEqual(const KUnitConversion__Value* self, const KUnitConversion__Value* other) {
    return (*self != *other);
}

bool KUnitConversion__Value_IsNull(const KUnitConversion__Value* self) {
    return self->isNull();
}

bool KUnitConversion__Value_IsValid(const KUnitConversion__Value* self) {
    return self->isValid();
}

double KUnitConversion__Value_Number(const KUnitConversion__Value* self) {
    return static_cast<double>(self->number());
}

KUnitConversion__Unit* KUnitConversion__Value_Unit(const KUnitConversion__Value* self) {
    return new KUnitConversion::Unit(self->unit());
}

libqt_string KUnitConversion__Value_ToString(const KUnitConversion__Value* self) {
    QString _ret = self->toString();
    // Convert QString from UTF-16 in C++ RAII memory to UTF-8 in manually-managed C memory
    QByteArray _b = _ret.toUtf8();
    libqt_string _str;
    _str.len = _b.length();
    _str.data = static_cast<const char*>(malloc(_str.len + 1));
    memcpy((void*)_str.data, _b.data(), _str.len);
    ((char*)_str.data)[_str.len] = '\0';
    return _str;
}

libqt_string KUnitConversion__Value_ToSymbolString(const KUnitConversion__Value* self) {
    QString _ret = self->toSymbolString();
    // Convert QString from UTF-16 in C++ RAII memory to UTF-8 in manually-managed C memory
    QByteArray _b = _ret.toUtf8();
    libqt_string _str;
    _str.len = _b.length();
    _str.data = static_cast<const char*>(malloc(_str.len + 1));
    memcpy((void*)_str.data, _b.data(), _str.len);
    ((char*)_str.data)[_str.len] = '\0';
    return _str;
}

KUnitConversion__Value* KUnitConversion__Value_Round(KUnitConversion__Value* self, unsigned int decimals) {
    KUnitConversion::Value& _ret = self->round(static_cast<uint>(decimals));
    // Cast returned reference into pointer
    return &_ret;
}

KUnitConversion__Value* KUnitConversion__Value_ConvertTo(const KUnitConversion__Value* self, const KUnitConversion__Unit* unit) {
    return new KUnitConversion::Value(self->convertTo(*unit));
}

KUnitConversion__Value* KUnitConversion__Value_ConvertTo2(const KUnitConversion__Value* self, int unit) {
    return new KUnitConversion::Value(self->convertTo(static_cast<KUnitConversion::UnitId>(unit)));
}

KUnitConversion__Value* KUnitConversion__Value_ConvertTo3(const KUnitConversion__Value* self, const libqt_string unit) {
    QString unit_QString = QString::fromUtf8(unit.data, unit.len);
    return new KUnitConversion::Value(self->convertTo(unit_QString));
}

libqt_string KUnitConversion__Value_ToString1(const KUnitConversion__Value* self, int fieldWidth) {
    QString _ret = self->toString(static_cast<int>(fieldWidth));
    // Convert QString from UTF-16 in C++ RAII memory to UTF-8 in manually-managed C memory
    QByteArray _b = _ret.toUtf8();
    libqt_string _str;
    _str.len = _b.length();
    _str.data = static_cast<const char*>(malloc(_str.len + 1));
    memcpy((void*)_str.data, _b.data(), _str.len);
    ((char*)_str.data)[_str.len] = '\0';
    return _str;
}

libqt_string KUnitConversion__Value_ToString2(const KUnitConversion__Value* self, int fieldWidth, char format) {
    QString _ret = self->toString(static_cast<int>(fieldWidth), static_cast<char>(format));
    // Convert QString from UTF-16 in C++ RAII memory to UTF-8 in manually-managed C memory
    QByteArray _b = _ret.toUtf8();
    libqt_string _str;
    _str.len = _b.length();
    _str.data = static_cast<const char*>(malloc(_str.len + 1));
    memcpy((void*)_str.data, _b.data(), _str.len);
    ((char*)_str.data)[_str.len] = '\0';
    return _str;
}

libqt_string KUnitConversion__Value_ToString3(const KUnitConversion__Value* self, int fieldWidth, char format, int precision) {
    QString _ret = self->toString(static_cast<int>(fieldWidth), static_cast<char>(format), static_cast<int>(precision));
    // Convert QString from UTF-16 in C++ RAII memory to UTF-8 in manually-managed C memory
    QByteArray _b = _ret.toUtf8();
    libqt_string _str;
    _str.len = _b.length();
    _str.data = static_cast<const char*>(malloc(_str.len + 1));
    memcpy((void*)_str.data, _b.data(), _str.len);
    ((char*)_str.data)[_str.len] = '\0';
    return _str;
}

libqt_string KUnitConversion__Value_ToString4(const KUnitConversion__Value* self, int fieldWidth, char format, int precision, const QChar* fillChar) {
    QString _ret = self->toString(static_cast<int>(fieldWidth), static_cast<char>(format), static_cast<int>(precision), *fillChar);
    // Convert QString from UTF-16 in C++ RAII memory to UTF-8 in manually-managed C memory
    QByteArray _b = _ret.toUtf8();
    libqt_string _str;
    _str.len = _b.length();
    _str.data = static_cast<const char*>(malloc(_str.len + 1));
    memcpy((void*)_str.data, _b.data(), _str.len);
    ((char*)_str.data)[_str.len] = '\0';
    return _str;
}

libqt_string KUnitConversion__Value_ToSymbolString1(const KUnitConversion__Value* self, int fieldWidth) {
    QString _ret = self->toSymbolString(static_cast<int>(fieldWidth));
    // Convert QString from UTF-16 in C++ RAII memory to UTF-8 in manually-managed C memory
    QByteArray _b = _ret.toUtf8();
    libqt_string _str;
    _str.len = _b.length();
    _str.data = static_cast<const char*>(malloc(_str.len + 1));
    memcpy((void*)_str.data, _b.data(), _str.len);
    ((char*)_str.data)[_str.len] = '\0';
    return _str;
}

libqt_string KUnitConversion__Value_ToSymbolString2(const KUnitConversion__Value* self, int fieldWidth, char format) {
    QString _ret = self->toSymbolString(static_cast<int>(fieldWidth), static_cast<char>(format));
    // Convert QString from UTF-16 in C++ RAII memory to UTF-8 in manually-managed C memory
    QByteArray _b = _ret.toUtf8();
    libqt_string _str;
    _str.len = _b.length();
    _str.data = static_cast<const char*>(malloc(_str.len + 1));
    memcpy((void*)_str.data, _b.data(), _str.len);
    ((char*)_str.data)[_str.len] = '\0';
    return _str;
}

libqt_string KUnitConversion__Value_ToSymbolString3(const KUnitConversion__Value* self, int fieldWidth, char format, int precision) {
    QString _ret = self->toSymbolString(static_cast<int>(fieldWidth), static_cast<char>(format), static_cast<int>(precision));
    // Convert QString from UTF-16 in C++ RAII memory to UTF-8 in manually-managed C memory
    QByteArray _b = _ret.toUtf8();
    libqt_string _str;
    _str.len = _b.length();
    _str.data = static_cast<const char*>(malloc(_str.len + 1));
    memcpy((void*)_str.data, _b.data(), _str.len);
    ((char*)_str.data)[_str.len] = '\0';
    return _str;
}

libqt_string KUnitConversion__Value_ToSymbolString4(const KUnitConversion__Value* self, int fieldWidth, char format, int precision, const QChar* fillChar) {
    QString _ret = self->toSymbolString(static_cast<int>(fieldWidth), static_cast<char>(format), static_cast<int>(precision), *fillChar);
    // Convert QString from UTF-16 in C++ RAII memory to UTF-8 in manually-managed C memory
    QByteArray _b = _ret.toUtf8();
    libqt_string _str;
    _str.len = _b.length();
    _str.data = static_cast<const char*>(malloc(_str.len + 1));
    memcpy((void*)_str.data, _b.data(), _str.len);
    ((char*)_str.data)[_str.len] = '\0';
    return _str;
}

void KUnitConversion__Value_Delete(KUnitConversion__Value* self) {
    delete self;
}

File: include/extras-kunitconversion/libvalue.zig
const QtC = @import("qt6zig");
const qtc = @import("qt6c");
const KUnitConversion__Unit = @import("libqt6").KUnitConversion__Unit;
const QChar = @import("libqt6").QChar;
const QVariant = @import("libqt6").QVariant;
const std = @import("std");
const unit_enums = @import("libunit.zig").enums;

/// ### [Upstream resources](https://api.kde.org/kunitconversion-value.html)
pub const KUnitConversion__Value = extern struct {
    /// ### [Upstream resources](https://api.kde.org/kunitconversion-value.html)
    ///
    /// The pointer to the underlying Qt C++ object
    ///
    ptr: QtC.KUnitConversion__Value,

    pub const _is_KUnitConversion__Value = {};

    /// ### DEPRECATED: Use `new` instead
    ///
    pub const New = new;

    /// Allocate a new KUnitConversion::Value object in C++ memory
    ///
    pub fn new() KUnitConversion__Value {
        return .{ .ptr = qtc.KUnitConversion__Value_new() };
    }

    /// ### DEPRECATED: Use `new2` instead
    ///
    pub const New2 = new2;

    /// Allocate a new KUnitConversion::Value object in C++ memory
    ///
    /// ## Parameter(s):
    ///
    /// ` _number: f64 `
    ///
    /// ` _unit: KUnitConversion__Unit `
    ///
    pub fn new2(_number: f64, _unit: anytype) KUnitConversion__Value {
        comptime _ = @TypeOf(_unit)._is_KUnitConversion__Unit;
        return .{ .ptr = qtc.KUnitConversion__Value_new2(@bitCast(_number), @ptrCast(_unit.ptr)) };
    }

    /// ### DEPRECATED: Use `new3` instead
    ///
    pub const New3 = new3;

    /// Allocate a new KUnitConversion::Value object in C++ memory
    ///
    /// ## Parameter(s):
    ///
    /// ` _number: f64 `
    ///
    /// ` unitString: []const u8 `
    ///
    pub fn new3(_number: f64, unitString: []const u8) KUnitConversion__Value {
        const unitString_str = qtc.libqt_string{
            .len = unitString.len,
            .data = unitString.ptr,
        };
        return .{ .ptr = qtc.KUnitConversion__Value_new3(@bitCast(_number), unitString_str) };
    }

    /// ### DEPRECATED: Use `new4` instead
    ///
    pub const New4 = new4;

    /// Allocate a new KUnitConversion::Value object in C++ memory
    ///
    /// ## Parameter(s):
    ///
    /// ` _number: f64 `
    ///
    /// ` unitId: unit_enums.UnitId `
    ///
    pub fn new4(_number: f64, unitId: i32) KUnitConversion__Value {
        return .{ .ptr = qtc.KUnitConversion__Value_new4(@bitCast(_number), @bitCast(unitId)) };
    }

    /// ### DEPRECATED: Use `new5` instead
    ///
    pub const New5 = new5;

    /// Allocate a new KUnitConversion::Value object in C++ memory
    ///
    /// ## Parameter(s):
    ///
    /// ` _number: QVariant `
    ///
    /// ` unitString: []const u8 `
    ///
    pub fn new5(_number: anytype, unitString: []const u8) KUnitConversion__Value {
        comptime _ = @TypeOf(_number)._is_QVariant;
        const unitString_str = qtc.libqt_string{
            .len = unitString.len,
            .data = unitString.ptr,
        };
        return .{ .ptr = qtc.KUnitConversion__Value_new5(@ptrCast(_number.ptr), unitString_str) };
    }

    /// ### DEPRECATED: Use `new6` instead
    ///
    pub const New6 = new6;

    /// Allocate a new KUnitConversion::Value object in C++ memory
    ///
    /// ## Parameter(s):
    ///
    /// ` other: KUnitConversion__Value `
    ///
    pub fn new6(other: anytype) KUnitConversion__Value {
        comptime _ = @TypeOf(other)._is_KUnitConversion__Value;
        return .{ .ptr = qtc.KUnitConversion__Value_new6(@ptrCast(other.ptr)) };
    }

    /// ### DEPRECATED: Use `operatorAssign` instead
    ///
    pub const OperatorAssign = operatorAssign;

    /// ### [Upstream resources](https://api.kde.org/kunitconversion-value.html#operator-eq)
    ///
    /// ## Parameter(s):
    ///
    /// ` self: KUnitConversion__Value `
    ///
    /// ` other: KUnitConversion__Value `
    ///
    pub fn operatorAssign(self: KUnitConversion__Value, other: anytype) void {
        comptime _ = @TypeOf(other)._is_KUnitConversion__Value;
        qtc.KUnitConversion__Value_OperatorAssign(@ptrCast(self.ptr), @ptrCast(other.ptr));
    }

    /// ### DEPRECATED: Use `swap` instead
    ///
    pub const Swap = swap;

    /// ### [Upstream resources](https://api.kde.org/kunitconversion-value.html#swap)
    ///
    /// ## Parameter(s):
    ///
    /// ` self: KUnitConversion__Value `
    ///
    /// ` other: KUnitConversion__Value `
    ///
    pub fn swap(self: KUnitConversion__Value, other: anytype) void {
        comptime _ = @TypeOf(other)._is_KUnitConversion__Value;
        qtc.KUnitConversion__Value_Swap(@ptrCast(self.ptr), @ptrCast(other.ptr));
    }

    /// ### DEPRECATED: Use `operatorEqual` instead
    ///
    pub const OperatorEqual = operatorEqual;

    /// ### [Upstream resources](https://api.kde.org/kunitconversion-value.html#operator-eq-eq)
    ///
    /// ## Parameter(s):
    ///
    /// ` self: KUnitConversion__Value `
    ///
    /// ` other: KUnitConversion__Value `
    ///
    pub fn operatorEqual(self: KUnitConversion__Value, other: anytype) bool {
        comptime _ = @TypeOf(other)._is_KUnitConversion__Value;
        return qtc.KUnitConversion__Value_OperatorEqual(@ptrCast(self.ptr), @ptrCast(other.ptr));
    }

    /// ### DEPRECATED: Use `operatorNotEqual` instead
    ///
    pub const OperatorNotEqual = operatorNotEqual;

    /// ### [Upstream resources](https://api.kde.org/kunitconversion-value.html#operator-not-eq)
    ///
    /// ## Parameter(s):
    ///
    /// ` self: KUnitConversion__Value `
    ///
    /// ` other: KUnitConversion__Value `
    ///
    pub fn operatorNotEqual(self: KUnitConversion__Value, other: anytype) bool {
        comptime _ = @TypeOf(other)._is_KUnitConversion__Value;
        return qtc.KUnitConversion__Value_OperatorNotEqual(@ptrCast(self.ptr), @ptrCast(other.ptr));
    }

    /// ### DEPRECATED: Use `isNull` instead
    ///
    pub const IsNull = isNull;

    /// ### [Upstream resources](https://api.kde.org/kunitconversion-value.html#isNull)
    ///
    /// ## Parameter(s):
    ///
    /// ` self: KUnitConversion__Value `
    ///
    pub fn isNull(self: KUnitConversion__Value) bool {
        return qtc.KUnitConversion__Value_IsNull(@ptrCast(self.ptr));
    }

    /// ### DEPRECATED: Use `isValid` instead
    ///
    pub const IsValid = isValid;

    /// ### [Upstream resources](https://api.kde.org/kunitconversion-value.html#isValid)
    ///
    /// ## Parameter(s):
    ///
    /// ` self: KUnitConversion__Value `
    ///
    pub fn isValid(self: KUnitConversion__Value) bool {
        return qtc.KUnitConversion__Value_IsValid(@ptrCast(self.ptr));
    }

    /// ### DEPRECATED: Use `number` instead
    ///
    pub const Number = number;

    /// ### [Upstream resources](https://api.kde.org/kunitconversion-value.html#number)
    ///
    /// ## Parameter(s):
    ///
    /// ` self: KUnitConversion__Value `
    ///
    pub fn number(self: KUnitConversion__Value) f64 {
        return qtc.KUnitConversion__Value_Number(@ptrCast(self.ptr));
    }

    /// ### DEPRECATED: Use `unit` instead
    ///
    pub const Unit = unit;

    /// ### [Upstream resources](https://api.kde.org/kunitconversion-value.html#unit)
    ///
    /// ## Parameter(s):
    ///
    /// ` self: KUnitConversion__Value `
    ///
    pub fn unit(self: KUnitConversion__Value) KUnitConversion__Unit {
        return .{ .ptr = qtc.KUnitConversion__Value_Unit(@ptrCast(self.ptr)) };
    }

    /// ### DEPRECATED: Use `toString` instead
    ///
    pub const ToString = toString;

    /// ### [Upstream resources](https://api.kde.org/kunitconversion-value.html#toString)
    ///
    /// ## Parameter(s):
    ///
    /// ` self: KUnitConversion__Value `
    ///
    /// ` allocator: std.mem.Allocator `
    ///
    pub fn toString(self: KUnitConversion__Value, allocator: std.mem.Allocator) []const u8 {
        var _str = qtc.KUnitConversion__Value_ToString(@ptrCast(self.ptr));
        defer qtc.libqt_string_free(&_str);
        const _ret = allocator.alloc(u8, _str.len) catch @panic("KUnitConversion__Value.toString: Memory allocation failed");
        @memcpy(_ret, _str.data[0.._str.len]);
        return _ret;
    }

    /// ### DEPRECATED: Use `toSymbolString` instead
    ///
    pub const ToSymbolString = toSymbolString;

    /// ### [Upstream resources](https://api.kde.org/kunitconversion-value.html#toSymbolString)
    ///
    /// ## Parameter(s):
    ///
    /// ` self: KUnitConversion__Value `
    ///
    /// ` allocator: std.mem.Allocator `
    ///
    pub fn toSymbolString(self: KUnitConversion__Value, allocator: std.mem.Allocator) []const u8 {
        var _str = qtc.KUnitConversion__Value_ToSymbolString(@ptrCast(self.ptr));
        defer qtc.libqt_string_free(&_str);
        const _ret = allocator.alloc(u8, _str.len) catch @panic("KUnitConversion__Value.toSymbolString: Memory allocation failed");
        @memcpy(_ret, _str.data[0.._str.len]);
        return _ret;
    }

    /// ### DEPRECATED: Use `round` instead
    ///
    pub const Round = round;

    /// ### [Upstream resources](https://api.kde.org/kunitconversion-value.html#round)
    ///
    /// ## Parameter(s):
    ///
    /// ` self: KUnitConversion__Value `
    ///
    /// ` decimals: u32 `
    ///
    pub fn round(self: KUnitConversion__Value, decimals: u32) KUnitConversion__Value {
        return .{ .ptr = qtc.KUnitConversion__Value_Round(@ptrCast(self.ptr), @bitCast(decimals)) };
    }

    /// ### DEPRECATED: Use `convertTo` instead
    ///
    pub const ConvertTo = convertTo;

    /// ### [Upstream resources](https://api.kde.org/kunitconversion-value.html#convertTo)
    ///
    /// ## Parameter(s):
    ///
    /// ` self: KUnitConversion__Value `
    ///
    /// ` _unit: KUnitConversion__Unit `
    ///
    pub fn convertTo(self: KUnitConversion__Value, _unit: anytype) KUnitConversion__Value {
        comptime _ = @TypeOf(_unit)._is_KUnitConversion__Unit;
        return .{ .ptr = qtc.KUnitConversion__Value_ConvertTo(@ptrCast(self.ptr), @ptrCast(_unit.ptr)) };
    }

    /// ### DEPRECATED: Use `convertTo2` instead
    ///
    pub const ConvertTo2 = convertTo2;

    /// ### [Upstream resources](https://api.kde.org/kunitconversion-value.html#convertTo)
    ///
    /// ## Parameter(s):
    ///
    /// ` self: KUnitConversion__Value `
    ///
    /// ` _unit: unit_enums.UnitId `
    ///
    pub fn convertTo2(self: KUnitConversion__Value, _unit: i32) KUnitConversion__Value {
        return .{ .ptr = qtc.KUnitConversion__Value_ConvertTo2(@ptrCast(self.ptr), @bitCast(_unit)) };
    }

    /// ### DEPRECATED: Use `convertTo3` instead
    ///
    pub const ConvertTo3 = convertTo3;

    /// ### [Upstream resources](https://api.kde.org/kunitconversion-value.html#convertTo)
    ///
    /// ## Parameter(s):
    ///
    /// ` self: KUnitConversion__Value `
    ///
    /// ` _unit: []const u8 `
    ///
    pub fn convertTo3(self: KUnitConversion__Value, _unit: []const u8) KUnitConversion__Value {
        const unit_str = qtc.libqt_string{
            .len = _unit.len,
            .data = _unit.ptr,
        };
        return .{ .ptr = qtc.KUnitConversion__Value_ConvertTo3(@ptrCast(self.ptr), unit_str) };
    }

    /// ### DEPRECATED: Use `toString1` instead
    ///
    pub const ToString1 = toString1;

    /// ### [Upstream resources](https://api.kde.org/kunitconversion-value.html#toString)
    ///
    /// ## Parameter(s):
    ///
    /// ` self: KUnitConversion__Value `
    ///
    /// ` allocator: std.mem.Allocator `
    ///
    /// ` fieldWidth: i32 `
    ///
    pub fn toString1(self: KUnitConversion__Value, allocator: std.mem.Allocator, fieldWidth: i32) []const u8 {
        var _str = qtc.KUnitConversion__Value_ToString1(@ptrCast(self.ptr), @bitCast(fieldWidth));
        defer qtc.libqt_string_free(&_str);
        const _ret = allocator.alloc(u8, _str.len) catch @panic("KUnitConversion__Value.toString1: Memory allocation failed");
        @memcpy(_ret, _str.data[0.._str.len]);
        return _ret;
    }

    /// ### DEPRECATED: Use `toString2` instead
    ///
    pub const ToString2 = toString2;

    /// ### [Upstream resources](https://api.kde.org/kunitconversion-value.html#toString)
    ///
    /// ## Parameter(s):
    ///
    /// ` self: KUnitConversion__Value `
    ///
    /// ` allocator: std.mem.Allocator `
    ///
    /// ` fieldWidth: i32 `
    ///
    /// ` format: u8 `
    ///
    pub fn toString2(self: KUnitConversion__Value, allocator: std.mem.Allocator, fieldWidth: i32, format: u8) []const u8 {
        var _str = qtc.KUnitConversion__Value_ToString2(@ptrCast(self.ptr), @bitCast(fieldWidth), @bitCast(format));
        defer qtc.libqt_string_free(&_str);
        const _ret = allocator.alloc(u8, _str.len) catch @panic("KUnitConversion__Value.toString2: Memory allocation failed");
        @memcpy(_ret, _str.data[0.._str.len]);
        return _ret;
    }

    /// ### DEPRECATED: Use `toString3` instead
    ///
    pub const ToString3 = toString3;

    /// ### [Upstream resources](https://api.kde.org/kunitconversion-value.html#toString)
    ///
    /// ## Parameter(s):
    ///
    /// ` self: KUnitConversion__Value `
    ///
    /// ` allocator: std.mem.Allocator `
    ///
    /// ` fieldWidth: i32 `
    ///
    /// ` format: u8 `
    ///
    /// ` precision: i32 `
    ///
    pub fn toString3(self: KUnitConversion__Value, allocator: std.mem.Allocator, fieldWidth: i32, format: u8, precision: i32) []const u8 {
        var _str = qtc.KUnitConversion__Value_ToString3(@ptrCast(self.ptr), @bitCast(fieldWidth), @bitCast(format), @bitCast(precision));
        defer qtc.libqt_string_free(&_str);
        const _ret = allocator.alloc(u8, _str.len) catch @panic("KUnitConversion__Value.toString3: Memory allocation failed");
        @memcpy(_ret, _str.data[0.._str.len]);
        return _ret;
    }

    /// ### DEPRECATED: Use `toString4` instead
    ///
    pub const ToString4 = toString4;

    /// ### [Upstream resources](https://api.kde.org/kunitconversion-value.html#toString)
    ///
    /// ## Parameter(s):
    ///
    /// ` self: KUnitConversion__Value `
    ///
    /// ` allocator: std.mem.Allocator `
    ///
    /// ` fieldWidth: i32 `
    ///
    /// ` format: u8 `
    ///
    /// ` precision: i32 `
    ///
    /// ` fillChar: QChar `
    ///
    pub fn toString4(self: KUnitConversion__Value, allocator: std.mem.Allocator, fieldWidth: i32, format: u8, precision: i32, fillChar: anytype) []const u8 {
        comptime _ = @TypeOf(fillChar)._is_QChar;
        var _str = qtc.KUnitConversion__Value_ToString4(@ptrCast(self.ptr), @bitCast(fieldWidth), @bitCast(format), @bitCast(precision), @ptrCast(fillChar.ptr));
        defer qtc.libqt_string_free(&_str);
        const _ret = allocator.alloc(u8, _str.len) catch @panic("KUnitConversion__Value.toString4: Memory allocation failed");
        @memcpy(_ret, _str.data[0.._str.len]);
        return _ret;
    }

    /// ### DEPRECATED: Use `toSymbolString1` instead
    ///
    pub const ToSymbolString1 = toSymbolString1;

    /// ### [Upstream resources](https://api.kde.org/kunitconversion-value.html#toSymbolString)
    ///
    /// ## Parameter(s):
    ///
    /// ` self: KUnitConversion__Value `
    ///
    /// ` allocator: std.mem.Allocator `
    ///
    /// ` fieldWidth: i32 `
    ///
    pub fn toSymbolString1(self: KUnitConversion__Value, allocator: std.mem.Allocator, fieldWidth: i32) []const u8 {
        var _str = qtc.KUnitConversion__Value_ToSymbolString1(@ptrCast(self.ptr), @bitCast(fieldWidth));
        defer qtc.libqt_string_free(&_str);
        const _ret = allocator.alloc(u8, _str.len) catch @panic("KUnitConversion__Value.toSymbolString1: Memory allocation failed");
        @memcpy(_ret, _str.data[0.._str.len]);
        return _ret;
    }

    /// ### DEPRECATED: Use `toSymbolString2` instead
    ///
    pub const ToSymbolString2 = toSymbolString2;

    /// ### [Upstream resources](https://api.kde.org/kunitconversion-value.html#toSymbolString)
    ///
    /// ## Parameter(s):
    ///
    /// ` self: KUnitConversion__Value `
    ///
    /// ` allocator: std.mem.Allocator `
    ///
    /// ` fieldWidth: i32 `
    ///
    /// ` format: u8 `
    ///
    pub fn toSymbolString2(self: KUnitConversion__Value, allocator: std.mem.Allocator, fieldWidth: i32, format: u8) []const u8 {
        var _str = qtc.KUnitConversion__Value_ToSymbolString2(@ptrCast(self.ptr), @bitCast(fieldWidth), @bitCast(format));
        defer qtc.libqt_string_free(&_str);
        const _ret = allocator.alloc(u8, _str.len) catch @panic("KUnitConversion__Value.toSymbolString2: Memory allocation failed");
        @memcpy(_ret, _str.data[0.._str.len]);
        return _ret;
    }

    /// ### DEPRECATED: Use `toSymbolString3` instead
    ///
    pub const ToSymbolString3 = toSymbolString3;

    /// ### [Upstream resources](https://api.kde.org/kunitconversion-value.html#toSymbolString)
    ///
    /// ## Parameter(s):
    ///
    /// ` self: KUnitConversion__Value `
    ///
    /// ` allocator: std.mem.Allocator `
    ///
    /// ` fieldWidth: i32 `
    ///
    /// ` format: u8 `
    ///
    /// ` precision: i32 `
    ///
    pub fn toSymbolString3(self: KUnitConversion__Value, allocator: std.mem.Allocator, fieldWidth: i32, format: u8, precision: i32) []const u8 {
        var _str = qtc.KUnitConversion__Value_ToSymbolString3(@ptrCast(self.ptr), @bitCast(fieldWidth), @bitCast(format), @bitCast(precision));
        defer qtc.libqt_string_free(&_str);
        const _ret = allocator.alloc(u8, _str.len) catch @panic("KUnitConversion__Value.toSymbolString3: Memory allocation failed");
        @memcpy(_ret, _str.data[0.._str.len]);
        return _ret;
    }

    /// ### DEPRECATED: Use `toSymbolString4` instead
    ///
    pub const ToSymbolString4 = toSymbolString4;

    /// ### [Upstream resources](https://api.kde.org/kunitconversion-value.html#toSymbolString)
    ///
    /// ## Parameter(s):
    ///
    /// ` self: KUnitConversion__Value `
    ///
    /// ` allocator: std.mem.Allocator `
    ///
    /// ` fieldWidth: i32 `
    ///
    /// ` format: u8 `
    ///
    /// ` precision: i32 `
    ///
    /// ` fillChar: QChar `
    ///
    pub fn toSymbolString4(self: KUnitConversion__Value, allocator: std.mem.Allocator, fieldWidth: i32, format: u8, precision: i32, fillChar: anytype) []const u8 {
        comptime _ = @TypeOf(fillChar)._is_QChar;
        var _str = qtc.KUnitConversion__Value_ToSymbolString4(@ptrCast(self.ptr), @bitCast(fieldWidth), @bitCast(format), @bitCast(precision), @ptrCast(fillChar.ptr));
        defer qtc.libqt_string_free(&_str);
        const _ret = allocator.alloc(u8, _str.len) catch @panic("KUnitConversion__Value.toSymbolString4: Memory allocation failed");
        @memcpy(_ret, _str.data[0.._str.len]);
        return _ret;
    }

    /// ### DEPRECATED: Use `delete` instead
    ///
    pub const Delete = delete;

    /// Delete this object from C++ memory
    ///
    /// ## Parameter:
    ///
    /// ` self: KUnitConversion__Value `
    ///
    pub fn delete(self: KUnitConversion__Value) void {
        qtc.KUnitConversion__Value_Delete(@ptrCast(self.ptr));
    }
};
